lib/fastly_nsq/listener.rb

```diff
@@ -1,117 +1,53 @@
 # frozen_string_literal: true
 
-require 'fastly_nsq/message'
-require 'fastly_nsq/manager'
-require 'fastly_nsq/safe_thread'
-require 'fastly_nsq/listener/config'
+class FastlyNsq::Listener
+  DEFAULT_PRIORITY = 0
+  DEFAULT_CONNECTION_TIMEOUT = 5 # seconds
 
-module FastlyNsq
-  class Listener
-    include FastlyNsq::SafeThread
+  attr_reader :preprocessor, :topic, :processor, :priority, :channel, :logger, :consumer
 
-    def self.listen_to(*args)
-      new(*args).go
-    end
+  def initialize(topic:, processor:, preprocessor: FastlyNsq.preprocessor, channel: FastlyNsq.channel, consumer: nil,
+                 logger: FastlyNsq.logger, priority: DEFAULT_PRIORITY, connect_timeout: DEFAULT_CONNECTION_TIMEOUT)
 
-    def initialize(topic:, processor:, channel: nil, consumer: nil, **options)
-      @consumer     = consumer || FastlyNsq::Consumer.new(topic: topic, channel: channel)
-      @done         = false
-      @logger       = options.fetch :logger, FastlyNsq.logger
-      @manager      = options[:manager] || FastlyNsq::Manager.new
-      @preprocessor = options[:preprocessor]
-      @processor    = processor
-      @thread       = nil
-      @topic        = topic
-    end
+    raise ArgumentError, "processor #{processor.inspect} does not respond to #call" unless processor.respond_to?(:call)
+    raise ArgumentError, "priority #{priority.inspect} must be a Integer" unless priority.is_a?(Integer)
 
-    def identity
-      {
-        consumer:     @consumer,
-        manager:      @manager,
-        preprocessor: @preprocessor,
-        processor:    @processor,
-        topic:        @topic,
-      }
-    end
+    @channel      = channel
+    @logger       = logger
+    @preprocessor = preprocessor
+    @processor    = processor
+    @topic        = topic
+    @priority     = priority
```

Comment:
if you had this as `@priority = priority || DEFAULT_PRIORITY` rather than having it in the arguments it could be passed `nil` and still get the default, which could be useful.

Reply:
If `nil` was passed in, it would fail the `Fixnum` validation above.